_ingest-pipelines/processors/community_id.md

```diff
@@ -0,0 +1,178 @@
+---
+layout: default
+title: Community ID
+parent: Ingest processors
+nav_order: 55
+---
+
+# Community ID processor
+
+The `community_id` processor is used to generate the community ID flow hash for network flow tuples. The community ID flow hash algorithm is defined in the [community ID specification](https://github.com/corelight/community-id-spec). The processor-generated hash value can be used to correlate all related network events so that you can filter the network flow data by the hash value or generate statistics by aggregating on the hash field. The processor supports TCP, UDP, SCTP, ICMP, and IPv6-ICMP network protocols. The SHA-1 hash algorithm is used to generate the hash value.
+
+The following is the `community_id` processor syntax:
+
+```json
+{
+  "community_id": {
+    "source_ip_field": "source_ip",
+    "source_port_field": "source_port",
+    "destination_ip_field": "destination_ip",
+    "destination_port_field": "destination_port",
+    "iana_protocol_number_field": "iana_protocol_number",
+    "source_port_field": "source_port",
+    "target_field": "community_id"
+  }
+}
+```
+{% include copy-curl.html %}
+
+## Configuration parameters
+
+The following table lists the required and optional parameters for the `community_id` processor.
+
+Parameter | Required/Optional | Description |
+|-----------|-----------|-----------|
+`source_ip_field`  | Required  | The name of the field containing the source IP address.  |
+`source_port_field`  | Optional  | The name of the field containing the source port address. If the network protocol is TCP, UDP or SCTP, the field is required. Otherwise, it is not required.|
+`destination_ip_field`  | Required  | The name of the field containing the destination IP address. |
+`destination_port_field`  | Optional  | The name of the field containing the destination port address. If the network protocol is TCP, UDP or SCTP, the field is required. Otherwise, it is not required. |
+`iana_protocol_number`  | Optional  | The name of the field containing the protocol number defined by the Internet Assigned Numbers Authority (IANA). The supported values are 1 (ICMP), 6 (TCP), 17 (UDP), 58 (IPv6-ICMP), and 132 (SCTP). |
+`protocol_field`  | Optional  | The name of the field containing the protocol name. If `iana_protocol_number` is not set, the field is required. Otherwise, it is not required. |
```

Comment:
```suggestion
`protocol_field`  | Optional  | The name of the field containing the protocol name. If `iana_protocol_number` is not set, then the field is required. Otherwise, it is not required. |
```